### PR TITLE
docs: clarify differential load doc

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
 https://angular.io/guide/deployment#differential-loading is confusing.

Issue Number:  https://github.com/angular/angular/issues/32978

## What is the new behavior?
Remove redundancy, edit and organize for clarity, conform to style guidelines for identifying examples/code snippets

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/deployment.md [hunk 1]
```diff
@@ -479,42 +479,40 @@ To maximize compatibility, you could ship a single bundle that includes all your
...
 * The second bundle contains code in the old ES5 syntax, along with all necessary polyfills. This results in a larger bundle size, but supports older browsers.
 
-This strategy allows you to continue to build your web application to support multiple browsers, but only load the necessary code that the browser needs.
-
 ### Differential builds
 
-The Angular CLI handles differential loading for you as part of the _build_ process for deployment.
-The `ng build` command produces the necessary bundles used for differential loading, based on your browser support requirements and compilation target.
+When you deploy using the Angular CLI build process, you can choose how and when to support differential loading.
+The [`ng build`](cli/build) command queries the browser configuration and the build `target` to determine if support for legacy browsers is required, and whether the build should produce the necessary bundles used for differential loading.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: Wrapping test that is a link in backticks makes it hard to distinguish it is a clickable link.
Consider removing the backticks.

File: aio/content/guide/deployment.md [hunk 2]
```diff
@@ -479,42 +479,40 @@ To maximize compatibility, you could ship a single bundle that includes all your
...
 
-The Angular CLI uses two configurations for differential loading:
+The following configurations to determine your requirements.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: Verb missing :grin:

File: aio/content/guide/deployment.md [hunk 3]
```diff
@@ -479,42 +479,40 @@ To maximize compatibility, you could ship a single bundle that includes all your
...
 </div>
 
-The CLI queries the Browserslist configuration, and checks the `target` to determine if support for legacy browsers is required.
-The combination of these two configurations determines whether multiple bundles are produced when you create a _build_.
-When you create a development build using [`ng build`](cli/build) and differential loading is enabled, the output produced is simpler and easier to debug, allowing you to rely less on sourcemaps of compiled code.
-When you create a production build using [`ng build --prod`](cli/build), the CLI uses the defined configurations above to determine the bundles to build for deployment of your application.
+For a development build, the output produced by `ng build` is simpler and easier to debug, allowing you to rely less on sourcemaps of compiled code.
 
-The `index.html` file is also modified during the build process to include script tags that enable differential loading. See the sample output below from the `index.html` file produced during a build using `ng build`.
+For a production build, your configuration determines which the bundles are created for deployment of your application.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: `which the bundles` --> `which of the bundles` (?)

File: aio/content/guide/deployment.md [hunk 4]
```diff
@@ -538,26 +536,22 @@ Each script tag has a `type="module"` or `nomodule` attribute. Browsers with nat
...
 
 Differential loading is supported by default with version 8 and later of the Angular CLI.
-For each application project in your workspace, you can configure how builds are produced based on the `browserslist` and `tsconfig.json` files in your application project.
+For each application project in your workspace, you can configure how builds are produced based on the `browserslist` and `tsconfig.json` configuration files in your application project.
 
-For a newly created Angular application, the default `browserslist` looks like this:
+For a newly created Angular application, legacy browsers such as IE 9-11 are ignored, and the compilation target is ES2015.
 
-```
+<code-example language="none" header="browserlist">
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: `browserlist` --> `browserslist`

**Review by gkalpak — COMMENTED** [review]
I left a couple of comments for obvious things/typos I noticed, but someone from `fw-docs-packaging` or the cli team should verify the content.

**Review by alan-agius4 — APPROVED** [review]
LGTM

**Pull request closed without merging**